modules/phase_field/src/kernels/ACInterfaceAniso1.C

```diff
@@ -0,0 +1,183 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ACInterfaceAniso1.h"
+
+registerMooseObject("PhaseFieldApp", ACInterfaceAniso1);
+
+template <>
+InputParameters
+validParams<ACInterfaceAniso1>()
+{
+  InputParameters params = validParams<Kernel>();
+  params.addClassDescription(
+      "Gradient energy Allen-Cahn Kernel where the derivative of interface parameter kappa"
+      "wrt the gradient of order parameter is considered");
+  params.addParam<MaterialPropertyName>("mob_name", "L", "The mobility used with the kernel");
+  params.addParam<MaterialPropertyName>("kappa_name", "kappa_op", "The kappa used with the kernel");
+  params.addParam<MaterialPropertyName>("dkappadgrad_etaa_name",
+                                        "dkappadgrad_etaa",
+                                        "The derivative of the kappa with respect to grad_etaa");
+  params.addParam<MaterialPropertyName>(
+      "d2kappadgrad_etaa_name",
+      "d2kappadgrad_etaa",
+      "The second derivative of the kappa with respect to grad_etaa");
+  params.addCoupledVar("args", "Vector of nonlinear variable arguments this object depends on");
+  params.addRequiredCoupledVar("etas", "All eta_i order parameters of the multiphase problem");
+  params.addParam<bool>("variable_L",
+                        true,
+                        "The mobility is a function of any MOOSE variable (if "
+                        "this is set to false L must be constant over the "
+                        "entire domain!)");
```

Comment:
I see quite a bit of copy/paste here. Please don't do that. Find a way to use inheritance instead. This may require factoring out a common base class.